src/box/error.cc

```diff
@@ -172,50 +172,66 @@ client_error_create(struct error *e, va_list ap)
 	error_vformat_msg(e, r->errdesc, ap_copy);
 	va_end(ap_copy);
 	for (int i = 0; i < r->errfields_count; i++) {
+		const char *name = r->errfields[i].name;
 		switch (r->errfields[i].type) {
 		case ERRCODE_FIELD_TYPE_CHAR: {
 			char buf[2] = {(char)va_arg(ap, int), '\0'};
-			error_set_str(e, r->errfields[i].name, buf);
+			if (name != NULL)
```

Comment:
Nit: I'd use a boolean variable here:

```C
if (set_payload)
        ...
```

Reply:
Done as part of work on previous comment.